sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -514,6 +436,33 @@ class ResolveSessionCatalog(
         parseSessionCatalogFunctionIdentifier("CREATE FUNCTION", catalog, ident)
       CreateFunctionCommand(database, function, className, resources, isTemp, ignoreIfExists,
         replace)
+
+    case alter: AlterTable => checkAlterTable(alter)
```

Comment:
Should we introduce new rule since this has nothing to do with resolving session catalog?

Reply:
Can we move this to `CheckAnalysis`? I think it's better to use the same error message `Cannot alter a view with ALTER TABLE. Please use ALTER VIEW instead`. We can update the v1 `AlterTableAddColumns` test to use the new error message, if it's not too many.

Reply:
OK. Sounds good. I will also update the migration guide since this is changing v1 command behavior.

The reason it's checked here is that  `CheckAnalysis` uses `foreachUp`, so `UnresolvedTableWithViewExists` is handled first before handling `AlterTable`. Any suggestion?

Reply:
Oh, you probably meant to check this before calling `checkAnalysis`. I will update it.